smtp/smtp.js

```diff
@@ -128,23 +213,42 @@ class SMTP extends EventEmitter {
 					options
 				)
 			);
-			return;
 		}
 
-		const connected = err => {
-			if (!err) {
-				log(`connected: ${this.host}:${this.port}`);
-
-				if (this.ssl && !this.tls) {
-					// if key/ca/cert was passed in, check if connection is authorized
-					if (typeof this.ssl != 'boolean' && !this.sock.authorized) {
-						this.close(true);
-						const msg = 'could not establish an ssl connection';
-						caller(callback, SMTPError(msg, SMTPError.CONNECTIONAUTH, err));
-					} else {
-						this._secure = true;
-					}
+		/**
+		 * @returns {void}
+		 */
+		const connected = () => {
+			log(`connected: ${this.host}:${this.port}`);
+
+			if (this.ssl && !this.tls) {
+				// if key/ca/cert was passed in, check if connection is authorized
+				if (
+					typeof this.ssl !== 'boolean' &&
+					this.sock instanceof TLSSocket &&
```

Comment:
`instanceof` check to make the typechecker happy (`authorized` isn't a property on `Socket`), though this could be a comment assertion instead (the pattern would be `!/** @type {import("tls").TLSSocket} */(this.sock).authorized` if we wanted to be strict, or `@type {*}` if we didn't care)